debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReader.java

```diff
@@ -408,12 +414,19 @@ protected void handleGtidEvent(Event event) {
         String gtid = gtidEvent.getGtid();
         gtidSet.add(gtid);
         source.startGtid(gtid, gtidSet.toString()); // rather than use the client's GTID set
+        ignoreDmlEventByGtidSource = false;
+        if (gtidDmlSourceFilter != null && gtid != null) {
+            String uuid = gtid.substring(0, gtid.indexOf(":"));
```

Comment:
Do the `gitd` strings need to be trimmed? Again, just depends on whether they are already properly formatted. I believe that the `GtidSet` class is trimming internally, just to make sure inadvertent whitespace doesn't mess things up.